packages/navigator/src/browser/navigator-model.spec.ts

```diff
@@ -0,0 +1,304 @@
+        mockFileNavigatorTree = sinon.createStubInstance(FileNavigatorTree);
+        mockWorkspaceService = sinon.createStubInstance(WorkspaceService);
+        mockFilesystem = sinon.createStubInstance(FileSystemNode);
+        mockLabelProvider = sinon.createStubInstance(LabelProvider);
+        mockFileSystemWatcher = sinon.createStubInstance(FileSystemWatcher);
+        mockILogger = sinon.createStubInstance(Logger);
+        mockTreeSelectionService = sinon.createStubInstance(TreeSelectionServiceImpl);
+        mockTreeExpansionService = sinon.createStubInstance(TreeExpansionServiceImpl);
+        mockTreeNavigationService = sinon.createStubInstance(TreeNavigationService);
+        mockTreeSearch = sinon.createStubInstance(TreeSearch);
+
+        testContainer = new Container();
+        testContainer.bind(FileNavigatorModel).toSelf().inSingletonScope();
+        testContainer.bind(OpenerService).toConstantValue(mockOpenerService);
+        testContainer.bind(FileNavigatorTree).toConstantValue(mockFileNavigatorTree);
+        testContainer.bind(WorkspaceService).toConstantValue(mockWorkspaceService);
+        testContainer.bind(FileSystem).toConstantValue(mockFilesystem);
+        testContainer.bind(LabelProvider).toConstantValue(mockLabelProvider);
+        testContainer.bind(FileSystemWatcher).toConstantValue(mockFileSystemWatcher);
+        testContainer.bind(ILogger).toConstantValue(mockILogger);
+        testContainer.bind(Tree).toConstantValue(mockFileNavigatorTree);
+        testContainer.bind(TreeSelectionService).toConstantValue(mockTreeSelectionService);
+        testContainer.bind(TreeExpansionService).toConstantValue(mockTreeExpansionService);
+        testContainer.bind(TreeNavigationService).toConstantValue(mockTreeNavigationService);
+        testContainer.bind(TreeSearch).toConstantValue(mockTreeSearch);
+
+        sinon.stub(mockWorkspaceService, 'onWorkspaceChanged').value(mockWorkspaceServiceEmitter.event);
+        sinon.stub(mockFileSystemWatcher, 'onFilesChanged').value(mockFileChangeEmitter.event);
+        sinon.stub(mockFileSystemWatcher, 'onDidMove').value(mockFileMoveEmitter.event);
+        sinon.stub(mockFileNavigatorTree, 'onChanged').value(mockTreeChangeEmitter.event);
+        sinon.stub(mockTreeExpansionService, 'onExpansionChanged').value(mockExpansionChangeEmitter.event);
+
+        setup();
+        navigatorModel = testContainer.get<FileNavigatorModel>(FileNavigatorModel);
+    });
+    afterEach(() => {
+        toRestore.forEach(res => {
+            res.restore();
+        });
+        toRestore.length = 0;
+    });
+
+    it('should update the root(s) on receiving a WorkspaceChanged event from the WorkspaceService', done => {
+        sinon.stub(navigatorModel, 'updateRoot').callsFake(() => {
+            done(); // This test would time out if updateRoot() is not called
+        });
+        mockWorkspaceServiceEmitter.fire([]);
+    }).timeout(2000);
```

Comment:
Is the `.timeout` necessary?

Reply:
yes, at least it is useful when we run it from the vsCode debugger, which starts a command that explicitly says "no time out"